packages/core/src/components/alert/alert.tsx

```diff
@@ -172,6 +172,7 @@ export class Alert extends AbstractPureComponent<AlertProps> {
         return (
             <Dialog
                 {...overlayProps}
+                role="alertdialog"
```

Comment:
We may also want to set [`aria-modal="true"` when setting this role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal)

---

We may also want to follow this guidance if adding this role

> The alert dialog must have at least one focusable control — such as Confirm, Close, and Cancel — and focus must be moved to that control when the alert dialog appears. Alertdialogs can have additional interactive controls such as text fields and checkboxes.

From my logging it appears focus is initially on the start of the dialog focus trap
<img width="1614" alt="Screenshot 2024-06-17 at 11 15 56 PM" src="https://github.com/palantir/blueprint/assets/14102129/875387d4-5dbd-472d-bf59-d9c894b9a185">

Reply:
Good call. Tied `aria-modal` to the `enforceFocus` prop in `Dialog`, these seemed to be one in the same.